src/test/java/com/neo4j/docker/utils/Neo4jVersionTest.java

```diff
@@ -0,0 +1,193 @@
+        Neo4jVersion newer = new Neo4jVersion( 3, 5, 0, "-11" );
+        Neo4jVersion older = new Neo4jVersion( 3, 5, 0, "-10" );
+
+        Assertions.assertFalse( newer.isOlderThan( older ) ,
+                     String.format( "Didn't detect that %s is newer than %s using isOlderThan", newer, older ) );
+        Assertions.assertFalse( older.isNewerThan( newer ) ,
+                     String.format( "Didn't detect that %s is older than %s using isNewerThan", older, newer ) );
+        Assertions.assertTrue( newer.isAtLeastVersion( older ),
+                    String.format( "Didn't detect that %s is at least %s using isAtLeastVersion", newer, older ) );
+        Assertions.assertTrue( newer.isEqual(older),
+                    String.format( "Didn't detect that %s is equal to %s using isEqual", newer, older ) );
+    }
+
+    private void testOlderVsNewer( Neo4jVersion newer, Neo4jVersion older )
+    {
+        // isOlderThan
+        Assertions.assertTrue( older.isOlderThan( newer ) ,
+                    String.format( "Didn't detect that %s is older than %s using isOlderThan", older, newer ) );
+        Assertions.assertFalse( newer.isOlderThan( older ) ,
+                     String.format( "Didn't detect that %s is newer than %s using isOlderThan", newer, older ) );
+
+        // isNewerThan
+        Assertions.assertTrue( newer.isNewerThan( older ),
+                    String.format( "Didn't detect that %s is newer than %s using isNewerThan", newer, older ) );
+        Assertions.assertFalse( older.isNewerThan( newer ) ,
+                     String.format( "Didn't detect that %s is older than %s using isNewerThan", older, newer ) );
+
+        // isAtLeastVersion
+        Assertions.assertTrue( newer.isAtLeastVersion( older ),
+                    String.format( "Didn't detect that %s is newer than %s using isAtLeastVersion", newer, older ) );
+        Assertions.assertFalse( older.isAtLeastVersion( newer ) ,
+                     String.format( "Didn't detect that %s is older than %s using isAtLeastVersion", older, newer ) );
+
+        Assertions.assertFalse( newer.isEqual(older),
+                                String.format( "%s incorrectly is equal to %s", newer, older ) );
+        Assertions.assertNotEquals( newer, older, String.format( "%s incorrectly is equal to %s", newer, older ) );
+    }
+
+    @Test
+    public void testIsOlderThan_sameReleaseReturnsFalse()
+    {
+        Neo4jVersion version = new Neo4jVersion( 3, 5, 1 );
+
+        Assertions.assertFalse( version.isOlderThan( version ) ,
+                     "A release should not be older than itself" );
+        Assertions.assertFalse( version.isNewerThan( version ) ,
+                     "A release should not be newer than itself" );
+    }
+
+    @Test
+    void testIsCalVer_2024()
+    {
+        Neo4jVersion version = new Neo4jVersion(2024, 9, 0);
+        Assertions.assertTrue(version.isCalVer(), version+" should be flagged as a calendar version");
```

Comment:
I don't think we need to make `isCalVer()` public, unless we need it in places other than tests. For docker I think the important thing would be that it converts from and to string correctly and that when compared with other version we get the expected result.